lib/Sema/CSDiag.cpp

```diff
@@ -3443,15 +3443,37 @@ Expr *FailureDiagnosis::typeCheckChildIndependently(
       if (FT->isAutoClosure())
         convertType = FT->getResult();
 
-    // Replace archetypes and type parameters with UnresolvedType.
-    convertType = replaceTypeParametersWithUnresolved(convertType);
-    convertType = replaceTypeVariablesWithUnresolved(convertType);
-    
+    bool shouldNullify = false;
```

Comment:
This is just `objectType->is<ArchetypeType>() || objectType->hasUnresolvedType()`